pkg/sql/codegen/attribute/attribute.go

```diff
@@ -158,3 +159,35 @@ func (d Dictionary) GenerateTableInHTML() string {
 	l = append(l, `</table>`)
 	return strings.Join(l, "\n")
 }
+
+// Update updates `d` by copyint from `other` key by key
+func (d Dictionary) Update(other Dictionary) Dictionary {
+	for k, v := range other {
+		d[k] = v
+	}
+	return d
+}
+
+// NewDictionary create a new Dictionary according to `estimator`
+func NewDictionary(estimator, prefix string) Dictionary {
+	var d = Dictionary{}
+	for param, doc := range PremadeModelParamsDocs[estimator] {
+		d[prefix+param] = &Description{Unknown, doc, nil}
+	}
+	return d
+}
+
+// PremadeModelParamsDocs stores parameters and documents of all known models
+var PremadeModelParamsDocs map[string]map[string]string
+
+func init() {
+	if err := json.Unmarshal([]byte(ModelParameterJSON), &PremadeModelParamsDocs); err != nil {
+		panic(err) // assertion
+	}
+	// The following parameters of canned estimators are already supported in the COLUMN clause.
+	for _, v := range PremadeModelParamsDocs {
+		delete(v, "feature_columns")
```

Comment:
`attribute.go` is not engine specific and these attributes seems only used in Tensorflow.

Reply:
They are used both in TensorFlow and PAI TensorFlow, that's why I put them here. Though `codegen/pai` shares part of `codegen/tensorflow` at the moment, they may diverge someday.